### ISSUE TITLE
Console Error - No matching speed found for direction.

### ISSUE THREAD

**Issue by jazzyisj** [aukedejong/lovelace-windrose-card#23]
HASS - v2023.5.3
Windrose: v0.11.0

![image](https://github.com/aukedejong/lovelace-windrose-card/assets/44064559/0d1b4a47-4ed0-4c56-bc5d-c5493d4a3726)

**Comment by aukedejong** [comment]
This means that the direction and speed measurements are not close enough in time. Default is 1second before or after.

You can try to increase direction_speed_time_diff config value.
Changing the matching-strategy can also help.

You don't see anything in the windrose graph?

**Comment by jazzyisj** [comment]
I was getting results in the windrose, but also tonnes of console errors.  I've changed my config, now I'm only getting a handful of errors.  I played with the settings a bit, this was the best I could do.

<img width="493" alt="image" src="https://github.com/aukedejong/lovelace-windrose-card/assets/44064559/601b518d-374c-42e3-8038-4ab1197719b6">

```
type: custom:windrose-card
title: Wind Direction
max_width: 400
windspeed_bar_full: true
wind_direction_entity: sensor.wind_bearing
wind_direction_unit: degrees
windspeed_entities:
  - entity: sensor.wind_speed
    name: Average
input_speed_unit: kph
output_speed_unit: kph
direction_speed_time_diff: 10
matching_strategy: speed-first
```

**Comment by tangix** [comment]
Got the same issue as @jazzyisj but the changes suggested doesn't work for me.
Is there anywhere in HA I can change what's going on?
<img width="1176" alt="image" src="https://github.com/aukedejong/lovelace-windrose-card/assets/1013635/889352be-25e4-43b9-b7ae-afd9f54e3a74">

**Comment by jazzyisj** [comment]
@tangix  Fooling around with the parameters I was able to greatly reduce the amount of errors in the console, but not eliminate them all together.

**Comment by aukedejong** [comment]
Hi. I'm working on a new version with breaking changes.  It support statistics data and I think it should fix these matching problems you have.
I also changed the logging, showing only one log entry about the matching results.

I'm planning to release somewhere next week.

You understand why you get these errors?

**Comment by tangix** [comment]
> You understand why you get these errors?

No. I get local weather data from a WS-3500 station that sends to Home Assistant. It looks like the data is not sent at the same time as shown:
<img width="512" alt="image" src="https://github.com/aukedejong/lovelace-windrose-card/assets/1013635/faf9044a-789e-4494-a2b8-5a7756dd143f">
The direction is sent much more often than wind speed:
<img width="578" alt="image" src="https://github.com/aukedejong/lovelace-windrose-card/assets/1013635/d814a985-1a52-4ff1-af19-01018db5aaaa">
<img width="571" alt="image" src="https://github.com/aukedejong/lovelace-windrose-card/assets/1013635/7ca11d49-3dc2-4c03-b3f0-a9b8924ca26c">

**Comment by aukedejong** [comment]
Hmm, your speed entity only gets updated twice an hour?  That is not very often.

I think it best to wait for the next version and try using HA statistics.
Matching code is rewritten.

Now I'm also thinking about an other matching strategy.
Something like this: for every direction measurement, look back in time for the last speed measurement. No matter how long ago, within the timeframe, the last measured one should be the one used.
Automatic select the direction or speed as leading, which one has the most measurements.

Thanks for letting me know, with these issues, I can improve things.

**Comment by jazzyisj** [comment]
Resolved with v1.0.0.  Thank you!
